### PR TITLE
Doc updates for CSIMigrationInTreeOff feature flag

### PR DESCRIPTION
**Pull request by ddebroy**
Doc update for CSIMigrationInTreeOff feature flag introduced in v1.17 part of CSI Migration.

/cc @davidz627 @leakingtapan

### CODE REVIEW

**Comment by sftim** [comment]
He @ddebroy. Is there a pull request to [kubernetes/kubernetes](https://github.com/kubernetes/kubernetes) that this matches against?

**Comment by davidz627** [comment]
@sftim https://github.com/kubernetes/kubernetes/pull/83098

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -312,6 +313,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
 - `CSIDriverRegistry`: Enable all logic related to the CSIDriver API object in csi.storage.k8s.io.
 - `CSIInlineVolume`: Enable CSI Inline volumes support for pods.
 - `CSIMigration`: Enables shims and translation logic to route volume operations from in-tree plugins to corresponding pre-installed CSI plugins
+- `CSIMigrationInTreeOff` : Disables registration of in-tree storage plugins migrated to CSI
```
Comment (davidz627) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Could you explain a little bit more it's interaction with `CSIMigrationX` flags. (in that it only disables those ones)

**Comment by tengqm** [comment]
/hold
Please feel free to cancel the hold when the upstream change is in.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -63,6 +63,7 @@ different Kubernetes components.
 | `CSIInlineVolume` | `false` | Alpha | 1.15 | 1.15 |
 | `CSIInlineVolume` | `true` | Beta | 1.16 | - |
 | `CSIMigration` | `false` | Alpha | 1.14 | |
+| `CSIMigrationInTreeOff`  |  false  | Alpha | 1.17 | |
```
Comment (kbhawkey) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: @ddebroy , do you think this entry should follow after `CSIMigrationGCE`

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md (hunk 1 above)
Comment (kbhawkey) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: Possibly maintain alphabetical order?

**Comment by davidz627** [comment]
These feature gates have been modified completely - @ddebroy could you update to current state? Thanks!

**Comment by ddebroy** [comment]
Updated with list of new feature flags and enhanced the descriptions. @davidz627 @leakingtapan @adisky @andyzhangx PTAL.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -331,11 +337,16 @@ Each feature gate is designed for enabling/disabling a specific feature:
...
 - `CSIMigration`: Enables shims and translation logic to route volume operations from in-tree plugins to corresponding pre-installed CSI plugins
-- `CSIMigrationAWS`: Enables shims and translation logic to route volume operations from the AWS-EBS in-tree plugin to EBS CSI plugin
-- `CSIMigrationAzureDisk`: Enables shims and translation logic to route volume operations from the Azure-Disk in-tree plugin to Azure Disk CSI plugin
-- `CSIMigrationAzureFile`: Enables shims and translation logic to route volume operations from the Azure-File in-tree plugin to Azure File CSI plugin
-- `CSIMigrationGCE`: Enables shims and translation logic to route volume operations from the GCE-PD in-tree plugin to PD CSI plugin
-- `CSIMigrationOpenStack`: Enables shims and translation logic to route volume operations from the Cinder in-tree plugin to Cinder CSI plugin
+- `CSIMigrationAWS`: Enables shims and translation logic to route volume operations from the AWS-EBS in-tree plugin to EBS CSI plugin. Supports falling back to in-tree EBS plugin if a node does not have EBS CSI plugin installed and configured. Requires CSIMigration feature flag enabled.
```
Comment (davidz627) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md]: sorry I didn't see this earlier, I think we should make it clear that the fallback only happens if the node doesn't have the `Feature Flags`. It has nothing to do with the install status. Can you submit a follow-up PR to update @ddebroy ?

**Pull request merged**